modin/data_management/functions/default_methods/groupby_default.py

```diff
@@ -25,7 +26,7 @@ class GroupBy:
     ]
 
     @classmethod
-    def validate_by(cls, by):
+    def validate_by(cls, by, groupby_args=None):
```

Comment:
What is still not clear to me is why we need to pass in a dictionary here and mutate it when we are defaulting to pandas. Can you explain this part? Why do we need to mutate the dictionary inplace?

Reply:
Because we can't pass `by` and `level` at the same time to pandas, it doesn't support that case. So we need to mutate arguments that we pass to pandas by combining `by` and `level` in a single argument.

Reply:
I thought they were combined in the API layer, though. Is that not happening?

Reply:
Hm, previously I tested it only with MultiIndex and it was raising exception, that's why I thought that they not support that arguments combination. However doing the same with regular index works just fine:
```python
>>> df # Non MultiIndex frame
  col1  col2
0    a     1
1    b     2
2    b     3
3    a     4
>>> df.groupby(by="col1", level=0)
<pandas.core.groupby.generic.DataFrameGroupBy object at 0x000001BDC30E9E88>
>>> df.index = pandas.MultiIndex.from_arrays([[0, 1, 2, 3]])
>>> df.groupby("col1", level=0)
TypeError: 'numpy.ndarray' object is not callable
```
So now it seems like a bug in pandas for me. I'm going to open an issue in pandas, but what is our move for now, until they've fixed it?

Reply:
Well, pandas guys reported that this is not a bug, it's expected behavior:
https://github.com/pandas-dev/pandas/issues/40378#issuecomment-797203893